include/osquery/sdk.h

```diff
@@ -54,15 +54,14 @@ REGISTER_INTERNAL(ExternalSQLPlugin, "sql", "sql");
  * module call-in well defined symbol, declare their SDK constraints, then
  * use the REGISTER_MODULE call within `initModule`.
  */
-#define REGISTER_EXTERNAL(type, registry, name)                            \
-  __attribute__((constructor)) static void type##ExtensionRegistryItem() { \
-    Registry::add<type>(registry, name);                                   \
-  }
-
-/// Helper macro to write the `initModule` symbol without rewrites.
-#define DECLARE_MODULE(name)        \
-  extern "C" void initModule(void); \
-  __attribute__((constructor)) static void declareModule()
+#define REGISTER_EXTERNAL(type, registry, name)                               \
+  struct type##ExtensionRegistryItem : public InitializerInterface {          \
+    type##ExtensionRegistryItem(void) { registerPlugin(this); }               \
+    const char *id() const override { return name_.c_str(); }                 \
+    void run() const override { Registry::add<type>(registry, name); }        \
+    std::string name_{std::string(registry) + "." + name};                    \
```

Comment:
Can we move this into a preprocessor-concat `const char*` within the `id()` return?